packages/app/src/cli/services/function/build.ts

```diff
@@ -186,6 +186,31 @@ function getESBuildOptions(
   return esbuildOptions
 }
 
+export async function runWasmOpt(modulePath: string) {
+  const wasmOpt = wasmOptBinary()
+  await downloadBinary(wasmOpt)
+
+  const wasmOptDir = dirname(wasmOptBinary().path)
+
+  const command = `node`
+  const args = [
+    // invoke the js-wrapped wasm-opt binary
+    wasmOptBinary().name,
+    modulePath,
+    // pass these options to wasm-opt
+    '-Oz',
+    '--enable-bulk-memory',
+    '--strip-debug',
+    // overwrite our existing module with the optimized version
+    '-o',
+    modulePath,
+  ]
+
+  outputDebug(`Wasm binary: ${wasmOptBinary().name}`)
+  outputDebug('Optimizing this wasm binary using wasm-opt.')
+  await exec(command, args, {cwd: wasmOptDir})
```

Comment:
What do we do with an error running the optimiser here? If these are usually user error, then throwing an AbortError with next steps would be good.

Reply:
We just output it to the console at the moment. Looks something like this (if I force it by messing up the optimizer invocation command)

![](https://screenshot.click/13-40-2u8tu-8t75e.png)

Is this sufficient? We don't currently use the AbortError with our other downloadables.
This also really isn't something that there'd be any user-error on, since we don't ask for the user's input for running this step.